### ISSUE TITLE
The lastest mac release is considerrably slower than 3.0

### ISSUE THREAD

**Issue by BlowaterNostr** [lumehq/lume#194]

**Comment by reyamir** [comment]
Hi, thank for your report.
Can you share me some recording?
In the `main` branch, I've also pushed some commit for improvement but not released yet

If you have time, can you help me test the `main` branch instead?
Just run

```
pnpm install -r
cd src-tauri && cargo install
```

Then
```
pnpm tauri dev
```

**Comment by reyamir** [comment]
Other issue I think cause slow performance is mandate bootstrap relays. Defaults, Lume will connect to these relays:
```
wss://relay.nostr.net
wss://relay.damus.io
wss://directory.yabu.me
```

If user is far from those relays location, the query will be very slow. So, I also working on bootstrap relays selection, user can choose bootstrap relays or input their own like noStrudel

**Comment by reyamir** [comment]
Hi @BlowaterNostr, I've pushed some new releases which also have some improvement on performance. can you help me retest it

**Comment by BlowaterNostr** [comment]
@reyamir Is it the 4.0.8 release?

**Comment by BlowaterNostr** [comment]
4.0.8 is faster on Apple Silicon than the previous release but still slower than v3. I can see that v4 has more features and you try to make a more guided experience. But the smoothness of v3 is very good. I feel it's a lost for v4 to not be as smooth.

**Comment by reyamir** [comment]
Yup, thank for your feedback. Can you help me record your experience, you can create new account for it. I want to figure how normal user experiences with lume.

**Comment by reyamir** [comment]
Yo, I've found some factors cause performance bottlenecks. The next build will probaly faster.

**Comment by BlowaterNostr** [comment]
FYI https://www.youtube.com/watch?v=m90d2tYCPt4

**Comment by reyamir** [comment]
> It feels much better. FYI https://www.youtube.com/watch?v=m90d2tYCPt4
> 
> Crushed once when I closed all the columns but I could not reproduce it.

wow, thank for your detail video. I will improve the UX when user close all columns, it should show some empty state instead of blank screen.

**Comment by reyamir** [comment]
Hi @BlowaterNostr
In thel latest release, I've made a lot of improvement in term of performance, the notable feature is Lume will always sync data via negentropy and query directly to local database instead of realy. Lume also support gossip model now.

I've also made a lot of changes for column system which based on NIP-51 instead of custom code, you can create multiple follow sets and interest sets now

The overall performance is also improve on macOS Sonoma too, thank to latest Safari Webview updates.

You can download and try new version here: https://github.com/lumehq/lume/releases/latest

**Comment by BlowaterNostr** [comment]
<img width="549" alt="image" src="https://github.com/user-attachments/assets/e3a69ca6-b391-4878-8f59-2b072e2d01c6">
I am waiting in this page for more than 5 minutes. You disabled dev mode so I can't debug for you in the release.

**Comment by reyamir** [comment]
Lume is syncing unnecessary data (like NIP-04 dm,...), I will make a change and release new version later today.
